hvcc/generators/ir2c/ir2c.py

```diff
@@ -223,9 +223,10 @@ def compile(clazz, hv_ir_path, static_dir, output_dir, externs, copyright=None):
             obj_id = x["id"]
             o = ir["objects"][obj_id]
             process_list.extend(ir2c.get_class(o["type"]).get_C_process(
-                o["type"],
                 x,
-                ir["objects"][obj_id]["args"]))
+                o["type"],
+                obj_id,
```

Comment:
Is there any of these that actually uses `obj_id` ?

Reply:
I can't find any, so why add it to the function signatures?

Reply:
I'm going to :-)

Reply:
You are now assuming that your future code will get accepted ;)

Reply:
I agree with you on `clazz` and doing some refactoring for the sake of making stuff uniform

Reply:
But for the sake of this particular PR, it doesn't make sense to add preliminary elements to the function signature, imo.

(if I would do this at work my colleagues would first laugh at me, then hate me :P)

Reply:
I didn't spell it out specifically, but I do have a need for `obj_id` in `get_C_process()`. Since each unique expr~ object will have a unique evaluation function, get_C_process will need the obj_id so that it can bind the unique function to the call site in `Heavy_heavy::process()`. I've dug around and thought about it quite a bit and can't think of another way to do it.

If you make an analogy to SignalMath.py, the only way that its `get_C_process()` knows which math function to call is because it is passed in via the `obj_type` parameter. `obj_id` would work the same for expr~ as obj_type works for SignalMath.py

I don't mean to be presumptuous, but not sure how to progress with the limitation of not having a unique identifier in `get_C_process()`.

I feel like some of the changes that seem to make sense anyways would be easier to deal with on their own rather than part of a much bigger commit. Additionally I'd like to have some conversation about these changes so that I don't make a bunch of changes to make expr work and then have a hell of time trying to explain/sell all the changes that I did to support it or at that point get opinions on how I should have done it.

I don't want to start adding stuff "just in case", but I think that expr is different enough (semantically and performance-wise) from all of the other objects that it warrants the change.

If there is another way to achieve the same goal I'm totally open to trying it.

cheers!

Reply:
True, small bites are easier to deal with. I currently have a bunch of tiny branches that each do some minimal self-contained things (also because they are just prototypes).

In general adding "will be useful later" variables and function signature changes can be confusing to review. Especially when those future changes that need it can take a while before being integrated .. and it's not sure that their implementation is even final.

Just want to make it easier to follow the progress on this. I'm definitely pro having more uniform function signatures and function calls, but premature optimization can also be a pitfall. Especially when future implementations are still uncertain.

Perhaps similarly this is why many of these functions had `objects` as an argument, even though this was never actually used in `get_C_process()`. Implemented for "future usage", but then never was ;)

I do think it's just a minor issue, and we could go at length to go back and forth. It would just help to set some ground rules on how to push certain changes. When it's not obvious why something that is not being used is added it just makes code review quite difficult.

Reply:
that is completely reasonable :-) Thank you for the guidance